l10n_vn_account_invoice_number/README.rst

```diff
@@ -0,0 +1,81 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+======================
+Account Invoice Number
+======================
+
+Vietnamese regulations (from tax authorities) requires to display 3 values on the reports of accounting transactions:
+
+The 3 values to display are:
+
+- Tax Invoice Number
+- Tax Invoice Form
+- Tax Invoice Series: The references of the series of invoice (the reference of the accounting booklet)
+
+In Odoo, the following types of records need to store those values:
+
+- Customer Invoice
+- Supplier Invoice
+
+In Odoo, we need to store this information whenever the issuer of the accounting document is in Vietnam, so this module adds a control when validating a customer/supplier invoice for a issuer based in Vietnam.
+
+Notes:
+- Issuer is the Supplier (or Parent Company of the supplier) for Supplier Invoices
+- Issuer is Company of the Invoice for Customer Invoices
+
+Usage
+=====
+
+To use this module, you need to create / validate any of these:
+ - Customer Invoice
+ - Supplier Invoice
+
+When validating, a control will ensure that the fields added by this module are being set.
```

Comment:
After thinking again about it, I thinks this control should not be done because red/legal invoices are often issued/received after the payment is done.

But in in Odoo the payment can only be register when the invoice is validated.

If you want to keep this control, I suggest that we make it optional.

Reply:
@jcdrubay: all our customer request the Red Invoice when they buy something. After check the invoice carefully (is that name correct, is that address correct...), they will decide to received that invoice or request to reissue the other. The payment is base on the deal, maybe cash immediately, maybe 30 days or 60 days.
with some big invoice, about 30.000.000VND, they must pay via bank. when they transfer the money, they will write the invoice number in the bank transfer description.
We do the same with our supplier.

Reply:
We will suggest soon some code to make it configurable so each company can choose if Tax Invoice Number must be input or not before validating the invoice.
For my business (in Vietnam) it is common that I issue or request payments before the red invoice is actually issued. This is especially true for invoices related to down-payments.

About the payment by bank transfer, the legal limit is 20mVND

Reply:
when you validate a invoice, the date is fix, couldn't modify. but the date on the Red invoice is in sequence. so is that a point to allow to modify these info after validate? In current version, it is free to modify after validate.

about the payment in your case, is that an Advance payment?

sometimes, our customer need to pay before get the goods. In that case, we just make a draft payment. then validate it later when validate the invoice. Or just validate the payment, keep open the amount of money, then reconcile with invoice later. not sure is that the right way to do?

Reply:
PR submitted https://github.com/fanha99/l10n-vietnam/pull/1